include/net/net_core.h

```diff
@@ -33,23 +33,31 @@ extern "C" {
 
 /* Network subsystem logging helpers */
 
-#if defined(NET_LOG_ENABLED)
-#if !defined(SYS_LOG_DOMAIN)
-#define SYS_LOG_DOMAIN "net"
-#endif /* !SYS_LOG_DOMAIN */
-
-#undef SYS_LOG_LEVEL
-#ifndef NET_SYS_LOG_LEVEL
-#define SYS_LOG_LEVEL CONFIG_SYS_LOG_NET_LEVEL
+#if !defined(LOG_LEVEL)
+#if defined(NET_LOG_LEVEL)
+#define LOG_LEVEL NET_LOG_LEVEL
 #else
-#define SYS_LOG_LEVEL NET_SYS_LOG_LEVEL
-#endif /* !NET_SYS_LOG_LEVEL */
-
-#define NET_DBG(fmt, ...) SYS_LOG_DBG("(%p): " fmt, k_current_get(), \
-				      ##__VA_ARGS__)
-#define NET_ERR(fmt, ...) SYS_LOG_ERR(fmt, ##__VA_ARGS__)
-#define NET_WARN(fmt, ...) SYS_LOG_WRN(fmt, ##__VA_ARGS__)
-#define NET_INFO(fmt, ...) SYS_LOG_INF(fmt,  ##__VA_ARGS__)
+#define LOG_LEVEL CONFIG_NET_LOG_LEVEL_DEFAULT
+#define NET_LOG_LEVEL LOG_LEVEL
+#endif /* NET_LOG_LEVEL */
+
+#if LOG_LEVEL > CONFIG_NET_LOG_LEVEL_MAX
+#undef LOG_LEVEL
+#define LOG_LEVEL CONFIG_NET_LOG_LEVEL_MAX
+#endif /* LOG_LEVEL > CONFIG_NET_LOG_LEVEL_MAX */
+#endif /* !LOG_LEVEL */
+
+#if defined(CONFIG_LOG) && defined(NET_LOG_LEVEL) && (NET_LOG_LEVEL > 0)
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER();
+
+#define NET_DBG(fmt, ...) LOG_DBG("(%p): %s: " fmt, k_current_get(), \
```

Comment:
What about "(%p): %s(): " ?

Reply:
Omitting () from function name saves two characters without sacrificing any readability. I am not sure about the print format atm, this is what we currently print with sys_log, but of course we could change that when printing with logger.